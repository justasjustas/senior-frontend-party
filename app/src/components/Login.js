import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { ApiUtil } from '../api/apiUtil.js';
import { setAuthorizationToken } from '../utils/authorization.js';
import '../sass/login.scss';

class Login extends Component {
  state = {
    username: '',
    password: '',
  };

  handleInput = event => {
    const { name, value } = event.target;
    this.setState({[name]: value});
  }

  handleSubmit = event => {
    event.preventDefault();
    this.login()
  }

  login = () => {
    ApiUtil('/tokens', 'POST',
      { 
        username: this.state.username,
        password: this.state.password,
      }
    ).then(response => {
      setAuthorizationToken(response);
      this.props.history.push('/servers');
    }).catch(err => console.error(err));
  }

  render() {
    const { username, password } = this.state;

    return (
      <div className="login-container row justify-content-center align-items-center">
        <form onSubmit={this.handleSubmit} className="login-form">
          <div className="login-form-logo-container">
            <img src="static/images/logotype-testio.png" alt="logo"/>
          </div>
          <div className="form-group input-group login-form-group">
          <object type="image/svg+xml" data="static/images/ico-username.svg">
          </object>
            <input
              type="text"
              className="form-control login-form-input"
              value={username}
              name="username"
              onChange={this.handleInput}
            />
          </div>
          <div className="form-group input-group login-form-group">
            <object type="image/svg+xml"data="static/images/ico-lock.svg">
            </object>
            <input
              type="password"
              className="form-control login-form-input"
              name="password"
              value={password}
              onChange={this.handleInput}
            />
          </div>
          <div className="form-group login-form-group">
            <button
              type="submit"
              className="btn login-form-btn"
              disabled={!username || !password}
            >Log In</button>
          </div>
        </form>   
      </div>
    );
  }
}

export default withRouter(Login);