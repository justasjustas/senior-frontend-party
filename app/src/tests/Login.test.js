import React from 'react';
import Login from '../components/Login';
import renderer from 'react-test-renderer';

it('Login renders correctly', () => {
  const tree = renderer
    .create(<Login.WrappedComponent />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});