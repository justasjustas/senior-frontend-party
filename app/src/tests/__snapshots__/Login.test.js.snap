// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<div
  className="login-container row justify-content-center align-items-center"
>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <div
      className="login-form-logo-container"
    >
      <img
        alt="logo"
        src="static/images/logotype-testio.png"
      />
    </div>
    <div
      className="form-group"
    >
      <div
        className="login-form-group"
      >
        <div
          className="login-form-group-input-prepend"
        >
          <object
            data="static/images/ico-username.svg"
            type="image/svg+xml"
          >
            icon
          </object>
        </div>
        <input
          className="form-control login-form-group-input"
          name="username"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <div
        className="login-form-group"
      >
        <div
          className="login-form-group-input-prepend"
        >
          <object
            data="static/images/ico-lock.svg"
            type="image/svg+xml"
          >
            icon
          </object>
        </div>
        <input
          className="form-control login-form-group-input"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      className="form-group login-form-group"
    >
      <button
        className="btn login-form-btn"
        disabled={true}
        type="submit"
      >
        Log In
      </button>
    </div>
  </form>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="login-container row justify-content-center align-items-center"
>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <div
      className="login-form-logo-container"
    >
      <img
        alt="logo"
        src="static/images/logotype-testio.png"
      />
    </div>
    <div
      className="form-group"
    >
      <div
        className="login-form-group"
      >
        <div
          className="login-form-group-input-prepend"
        >
          <object
            data="static/images/ico-username.svg"
            type="image/svg+xml"
          >
            icon
          </object>
        </div>
        <input
          className="form-control login-form-group-input"
          name="username"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <div
        className="login-form-group"
      >
        <div
          className="login-form-group-input-prepend"
        >
          <object
            data="static/images/ico-lock.svg"
            type="image/svg+xml"
          >
            icon
          </object>
        </div>
        <input
          className="form-control login-form-group-input"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      className="form-group login-form-group"
    >
      <button
        className="btn login-form-btn"
        disabled={true}
        type="submit"
      >
        Log In
      </button>
    </div>
  </form>
</div>
`;
